adjuster/finagle/src/main/java/zipkin/sparkstreaming/adjuster/finagle/FinagleIssue343Adjuster.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2017 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.sparkstreaming.adjuster.finagle;
+
+import com.google.auto.value.AutoValue;
+import zipkin.Annotation;
+import zipkin.BinaryAnnotation;
+import zipkin.Span;
+import zipkin.internal.Util;
+import zipkin.sparkstreaming.Adjuster;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Contains adjustments for finagle issue 343 (https://github.com/twitter/finagle/issues/343).
+ * If applied, this adjuster drops finagle.flush annotation.
+ */
+@AutoValue
+public abstract class FinagleIssue343Adjuster extends Adjuster{
+
+  public static FinagleIssue343Adjuster.Builder newBuilder() {
+    return new AutoValue_FinagleIssue343Adjuster.Builder()
+        .adjustIssue343(false);
+  }
+
+  abstract boolean adjustIssue343();
+
+  @AutoValue.Builder
+  public interface Builder {
+    /**
+     * This adjuster handles a bug in finagle memcached library where the duration of a span ends
+     * up being more than the actual value because a span is submitted twice.
+     *
+     * The fix is to drops finagle.flush annotation for memcache spans. We look for "Hit" or "Miss"
+     * binary annotations to make sure the span is coming from memcache.
+     *
+     * For more details see
+     * https://github.com/twitter/finagle/issues/343
+     *
+     * Default false
+     */
+    Builder adjustIssue343(boolean adjustIssue343);
+
+    FinagleIssue343Adjuster build();
+  }
+
+  @Override protected boolean shouldAdjust(Span span) {
+    if (adjustIssue343()) {
+      boolean containsFinagleFlushAnnotation = false;
+      for (Annotation a : span.annotations) {
+        if (a.value.equals("finagle.flush")) {
+          containsFinagleFlushAnnotation = true;
+          break;
+        }
+      }
+      if (containsFinagleFlushAnnotation) {
+        for (BinaryAnnotation b : span.binaryAnnotations) {
```

Comment:
Please consider refactoring this into a static method also. Then this method becomes 
return checkContainsFinagleFlushAnnotation(span) && isMemcacheHit(span);